tox.ini

```diff
@@ -2,58 +2,64 @@
 mintoxversion = 2.3
 envlist = check_readme,flake8,covclean,py38,coverage,integrationtests
 
+# Settings defined in the top-level testenv section are automatically inherited by individual
+# environments unless overridden
 [testenv]
+basepython = python3.8
 passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH SOLC_BINARY INTEGRATION_TESTS_REPO
+allowlist_externals =
+    git
+    /bin/rm
+    /bin/ln
+    bash
 deps =
     pip-tools
     coverage
+commands_pre =
+    pip install --upgrade pip
+    pip install -rrequirements/tests.txt
 commands =
-    pip-sync requirements/tests.txt
     pip install -e .
     coverage run -m pytest  --random-order {posargs:tests}
 
 [testenv:covclean]
-basepython = python3.8
 skip_install = true
-deps = coverage
 commands = coverage erase
 
 [testenv:coverage]
-basepython = python3.8
 skip_install = true
-deps = coverage
 commands =
     coverage combine
     coverage report
 
+[testenv:codecoverage]
```

Comment:
I'd remove this `codecoverage` environment and just add everything inside `coverage`. I don't think we ever need to generate coverage reports without running pytest.

Reply:
Sorry, I just noticed that the `coverage` env didn't run pytest. Why do you think we need this `codecoverage` env?

Reply:
I'm not sure TBH. The testenv was already there I just moved it up to be together with coverage testenv as they are "similar".

Reply:
I agree with you 👍